src/app/multiselect/multiselect-dropdown.html

```diff
@@ -27,13 +27,15 @@
                     ng-click="multiselectDropdown.deselectAll()">
                 </a>
             </li>
-            <li class="divider"></li>
-            <li ng-repeat="option in multiselectDropdown.optionsFiltered = (multiselectDropdown.options | filter : multiselectDropdown.search)">
+            <li ng-bind="group" ng-if="multiselectDropdown.isGrouped && !multiselectDropdown.isGroupEmpty(group)" ng-repeat-start="group in multiselectDropdown.groups"></li>
+            <li class="divider" ng-if="!multiselectDropdown.isGrouped"></li>
```

Comment:
I would probably just always include the divider before the outer `ngRepeat`. Also, we can throw a `dropdown-header` class on the group header list item.